randovania/generator/filler/player_state.py

```diff
@@ -248,6 +248,13 @@ def should_have_hint(self, pickup: PickupEntry, current_uncollected: Uncollected
 
         return can_hint
 
+    def count_self_locations(self, locations: WeightedLocations) -> int:
+        count = 0
+        for player, index in locations.keys():
+            if player == self:
+                count += 1
+        return count
```

Comment:
any reason this shouldn't be `return sum(1 for player, _ in locations.keys() if player == self)`?

Reply:
tired brain couldn't confirm they were equivalent